src/fairchem/core/datasets/ase_datasets.py

```diff
@@ -123,15 +115,15 @@ def __getitem__(self, idx):
             )
 
         sid = atoms.info.get("sid", self.ids[idx])
-        fid = atoms.info.get("fid", tensor([0]))
 
-        # Convert to data object
-        data_object = self.a2g.convert(atoms, sid)
-        data_object.fid = fid
-        data_object.natoms = len(atoms)
+        fid = atoms.info.get("fid", None)
+        if fid is not None:
+            sid = f"{sid}-frame-{fid}"
+
+        data_object = self.a2g(atoms, sid=sid)
 
         # apply linear reference
-        if self.a2g.r_energy is True and self.lin_ref is not None:
+        if self.lin_ref is not None:
             data_object.energy -= sum(self.lin_ref[data_object.atomic_numbers.long()])
```

Comment:
are we still using this in V2?

Reply:
good catch, we're not using afaik. It would be a good idea to remove it

Reply:
I made a note of this and will open a new PR for it tomorrow. Its probably straightforward, but would rather keep as much separate from this spiraling PR as possible.